src/source/pausable.rs

```diff
@@ -4,26 +4,46 @@ use Sample;
 use Source;
 
 /// Internal function that builds a `Pausable` object.
-pub fn pausable<I>(source: I, paused: bool) -> Pausable<I> {
+pub fn pausable<I>(source: I, paused: bool) -> Pausable<I>
+where
+    I: Source,
+    I::Item: Sample,
+{
+    let paused_channels = if paused {
+        Some(source.channels())
+    } else {
+        None
+    };
     Pausable {
         input: source,
-        paused: paused,
+        paused_channels: paused_channels,
+        remaining_paused_samples: 0,
     }
 }
 
 #[derive(Clone, Debug)]
 pub struct Pausable<I> {
     input: I,
-    paused: bool,
+    paused_channels: Option<u16>,
+    remaining_paused_samples: u16,
 }
 
-impl<I> Pausable<I> {
+impl<I> Pausable<I>
+where
+    I: Source,
+    I::Item: Sample,
+{
     /// Sets whether the filter applies.
     ///
     /// If set to true, the inner sound stops playing and no samples are processed from it.
     #[inline]
     pub fn set_paused(&mut self, paused: bool) {
-        self.paused = paused;
+        // Minimize calls to channels by only calling on state change
```

Comment:
This comment is wrong. You always call the function when paused is true, even if it was true before. In general, `set_paused` shouldn't be called often so minimizing calls shouldn't be needed. I'd say you should just remove the comment.